### PR TITLE
feat: add Slide, add classes prop to LinearProgress

### PR DESCRIPTION
**Pull request by javamonn**
- Adds the [`Slide`](https://material-ui-next.com/api/slide/) component, but does not include the props that the underlying `Transition` would use.
- Adds the `classes` prop to the existing `LinearProgress` component.

### CODE REVIEW

File: src/MaterialUI.re [hunk 1]
```diff
@@ -1309,6 +1311,42 @@ module Select = {
...
+      | Up => "up"
+      | Down => "down"
+      | Left => "left"
+      | Right => "right";
+  };
+  let make =
+      (
+        ~in_: option(bool)=?,
+        ~direction_: option(Direction.t)=?,
+        ~timeout: option({. "enter": float, "exit": float})=?,
```
Comment (bernalrs) [inline src/MaterialUI.re:1333]: The docs specify you should be able to provide a single float (or number in js) value as well.

File: src/MaterialUI.re [hunk 2]
```diff
@@ -1309,6 +1311,42 @@ module Select = {
...
+        children
+      ) =>
```
Comment (bernalrs) [inline src/MaterialUI.re:1335]: There are also some events missign which have been omitted from the docs.

  onEnter?: TransitionCallback, 
  onEntering?: TransitionCallback,
  onEntered?: TransitionCallback,
  onExit?: TransitionCallback,
  onExiting?: TransitionCallback,
  onExited?: TransitionCallback,

**Review by bernalrs — CHANGES_REQUESTED** [review]
Few comment inline.

File: src/MaterialUI.re (hunk 2 above)
Comment (bernalrs) [reply src/MaterialUI.re:1335]: Should also include the properties of the Transition class:

  appear?: boolean;
  unmountOnExit?: boolean;

**Comment by InsidersByte** [comment]
this will be added in #55

**Pull request closed without merging**